### DISCUSSION TITLE
The impact of Favorites on performance

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#601] [Q&A] [answered]
Hi. The fact that the number of displayed Favorites affects the map's performance is obvious and understandable. But can Favorites excluded from display and navigation (and only included in the collection as inactive) also affect the speed of map/location display?

During yesterday's trip, I encountered an unpleasant map problem. During navigation, the position cursor – and map – moved intermittently, every **dozen or so seconds**! At first I thought it was a location service issue, but changing Android <=> Google didn't make a difference. The rest of the voice messages regarding the distance were correct, and the zoom before the maneuvers took place at the right time, so location was not a problem. But on the other hand, the trace is torn and "angular". https://gpsjam.org did not show the influence of "neighbors" (which unfortunately happens). I tried to disable other services running in the background, but being on the road it was difficult to do tests.

It is true that I have the Kalman filter and smooth panning turned on, but I have always had it like this and I have never observed such behavior of the map! The only thing that has changed is the additional over a thousand Favorites in the next category. So I turned off display/navigation for most Favorites in the collection, but the map issue still persisted. 

Hence my question: can the very large number of Favorites in the collection, even when they are turned off, affect performance?

**Comment by devemux86** [comment, marked as answer]
Map layers (like the favorites) can affect the map performance **only** when they are visible.

Specifically the favorites can be hidden, but still participate in navigation.

> thousand Favorites

Do you really need all of them visible?

If not having "too many" favorites visible (and unnecessary for navigation),
then the problem could be something else?

The best in such cases is to restart the device, as Android likes to cache things.

Also disable Kalman filter, I will see if we really need it in next Cruiser.

In any case Cruiser advances to a new map engine where everything is implemented differently
and we will need to check them again. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> Map layers (like the favorites) can affect the map performance only when they are visible.

Thanks for the quick - as usual - answer 👍 I thought so, but I wanted to make sure.

> Do you really need all of them visible?

Only when planning a day trip. While driving, of course, this can be disturbing. That's why I have a lot of categories enabled only during navigation.

> The best in such cases is to restart the device, as Android likes to cache things.

Yes. I'm aware of this, but didn't want to do it during the tour, due to other background processes (which never affected map performance). I limited myself to restarting Cruiser, which did not help.

> new map engine where everything is implemented differently and we will need to check them again

I'm looking forward to it and will be happy to test it. Thank you for all this! 🙂

**Reply by devemux86** [reply]
> Only when planning a day trip. While driving, of course, this can be disturbing. That's why I have a lot of categories enabled only during navigation.

We might need to improve this workflow to be somehow automatic and avoid the manual turn on / off.

Currently we have 2 on/off switches:
- "Show on map" that works in all modes: route planning and navigation.
  - While we may need some groups visible only in route planning or navigation?

- "Use in navigation" allows a group to be used in navigation's nearby favorites
and works independently from the visibility.

**Reply by Ciutnik** [reply]
> While we may need some groups visible only in route planning or navigation?

Oh yes, that would be great (I dreamed of something like that on the forum a long time ago), but it's an additional complexity...

And back to the problem. Yesterday was terribly hot, maybe the device simply overheated and the processor slowed down to a minimum. Some time ago (once) I saw a message that "Cruiser had caused the device to overheat and therefore was closed!" 😲 It was a case when I forgot to turn off the application and it worked for a few hours at home, in my backpack.

Yesterday, it's good that the voice messages were on time and guided me flawlessly. Maybe the panels worked properly too, but I don't remember now.

Another trip in the evening, we'll see how everything will work (by the way, I'm complaining about the lack of automatic night map mode 😉)

**Reply by devemux86** [reply]
> It was a case when I forgot to turn off the application and it worked for a few hours at home, in my backpack.

Do you remember at what state? Navigation, GPS recording, something else?

**Reply by Ciutnik** [reply]
> Do you remember at what state? Navigation, GPS recording, something else?

Unfortunately, I don't remember the circumstances. It was in April. Probably the navigation was not active, because I would have heard messages about leaving the route, maybe following. I took the tablet out of my backpack and saw a message (I took a screenshot), translated as follows: 
"_**Cruiser app closed.** The Cruiser app was closed because it was causing the tablet to overheat. You can turn it back on later when the tablet has cooled down._ [OK]"

I have never seen anything like this before or since.

**Reply by ramondevic** [reply]
> > Only when planning a day trip. While driving, of course, this can be disturbing. That's why I have a lot of categories enabled only during navigation.
> 
> We might need to improve this workflow to be somehow automatic and avoid the manual turn on / off.
> 
> Currently we have 2 on/off switches:
> 
>     * "Show on map" that works in all modes: route planning and navigation.
>       
>       * While we may need some groups visible only in route planning or navigation?
> 
>     * "Use in navigation" allows a group to be used in navigation's nearby favorites
>       and works independently from the visibility.

I have quite a few groups, some of them are always activated when viewing the map and in navigation or whatever, and others I activate/deactivate depending on whether I need them for the route I'm taking.

I find it perfect as it is right now. :-)

**Reply by ramondevic** [reply]
Activating more or less layers on the map, e.g.: mountain passes, mountain peaks, viewpoints, tourist attractions, ... can penalize the performance of the application?

**Reply by devemux86** [reply]
> Activating more or less layers on the map, e.g.: mountain passes, mountain peaks, viewpoints, tourist attractions, ... can penalize the performance of the application?

No, these are layers embedded in the offline maps, so they are optimized for performance.

The overlays we draw on top of the map, like too many favorites or long tracks can influence the map performance.

So the advice is to avoid loading the map with too many overlays, just keep only the necessary ones visible! :slightly_smiling_face:

**Comment by devemux86** [comment]
> I'm complaining about the lack of automatic night map mode

@MotoUKRider prepared an excellent new theme (light / dark) for Cruiser 5,
so after this we can see again such various "quality of life" features. :slightly_smiling_face: 

![cruiser-mapsforge-preview-3](https://github.com/mapsforge/mapsforge/assets/125696109/61ec1d3e-f056-4297-af73-53a3f341e4fb)

**Reply by MotoUKRider** [reply]
@Ciutnik To add to @devemux86 tease of the new themes,  "Dark" has existed in the OpenGL/VTM style for some time and I also found the lack of auto switch (from the phone's light sensor) frustrating.

My solution, was to make a 2nd Profile in Cruiser to handle Day/Night riding. This offers **TWO** benefits: 1) It's easier/quicker/safer to switch a Profile than change the Map Style... 2) Other colours like the info Panels also needed to be "night friendly".

Here's what I came up with:
![cruiser-day-night](https://github.com/user-attachments/assets/57469c24-5963-4d88-a008-2de61245df28)

**Reply by Ciutnik** [reply]
@MotoUKRider Indeed, you are right! It's not just a question of the colour scheme of the map but of the whole application interface, led by the panels! 
It certainly concerns many users. I actually like the dark theme and panels even during the day. Something for everyone 🙂

**Reply by airald** [reply]
I've been doing the same since profiles exists. Before that I just change map colour scheme.
I change profile using Automate and a custom button BT keyboard.

**Reply by Ciutnik** [reply]
> I change profile using Automate

Interesting idea. Have you tried automating it based on the ambient brightness sensor?  It should work.

**Reply by airald** [reply]
Not yet, I think I preffer do it manually, I just need a double click on one of my handlebar buttons.

**Reply by Ciutnik** [reply]
I played with Tasker for a while. Yes, automatic profile change based on a light sensor can work! However... currently it probably doesn't make sense 😁

https://github.com/user-attachments/assets/503a882a-d125-4775-b954-c4a83069e2ff
